File: frontend/lib/planner/models/task_status.dart
/// Task lifecycle statuses, mirroring the backend planner domain enums.
enum TaskStatus {
  todo,
  inProgress,
  completed,
  cancelled;

  static TaskStatus fromWire(String? value) {
    switch (value?.toUpperCase()) {
      case 'IN_PROGRESS':
        return TaskStatus.inProgress;
      case 'COMPLETED':
        return TaskStatus.completed;
      case 'CANCELLED':
        return TaskStatus.cancelled;
      default:
        return TaskStatus.todo;
    }
  }

  String get wireName => switch (this) {
        TaskStatus.todo => 'TODO',
        TaskStatus.inProgress => 'IN_PROGRESS',
        TaskStatus.completed => 'COMPLETED',
        TaskStatus.cancelled => 'CANCELLED',
      };

  bool get isActive =>
      this == TaskStatus.todo || this == TaskStatus.inProgress;
}

File: backend/src/main/java/com/blistra/planner/domain/TaskStatus.java
package com.blistra.planner.domain;

/**
 * Lifecycle state of a planner task. Overdue is derived from due date/time and
 * the active state, never stored as a status.
 */
public enum TaskStatus {
    TODO,
    IN_PROGRESS,
    COMPLETED,
    CANCELLED
}